android/src/main/kotlin/com/mparticle/mparticle_flutter_sdk/MparticleFlutterSdkPlugin.kt

```diff
@@ -135,6 +138,69 @@ class MparticleFlutterSdkPlugin: FlutterPlugin, MethodCallHandler {
           jsonObj.put(entry.key.toString(), value)
         }.let { result.success((it ?: JSONObject()).toString()) }
       "logCommerceEvent" -> this.logCommerceEvent(call, result)
+      "getGDPRConsentState" -> getUser(call, result)?.let { user ->
+        user.consentState.gdprConsentState
+          .toList()
+          .fold(JSONObject()) { json, (key, value) ->
+            json.put(key, value.toJSON())
+          }
+          .let {
+            result.success(it.toString())
+          }
+      }?: result.error(TAG, "User not found", null)
+      "addGDPRConsentState" -> getUser(call, result)?.let { user ->
+        val consentState = call.toGDPRConsent(result)
+        if (consentState == null) {
+          result.error(TAG, "ConsentState should not be null", null)
+          return
+        }
+        ConsentState.withConsentState(user.consentState)
+          .addGDPRConsentState(consentState.first, consentState.second)
+          .build()
+          .let {
+            user.setConsentState(it)
+          }
+        result.success(true)
```

Comment:
should this `result.success(true)` be inside the `.let` above?

Reply:
in this case, it doesn't really matter. since `let` isn't null-protected (I think that's the right term...lacks a `?`), it is going to run every time this block executes. whether the `result.success()` call is where it is or inside `let`, it will be the exact same logic at runtime, so I thought organizationally it might a tiny it more sense to keep it outside the `let` and leave the previous chained calls purely "consent-setting stuff"